services/viva-ms/src/lambdas/submitCompletion.js

```diff
@@ -131,9 +130,15 @@ export async function main(event, context) {
     return false;
   }
 
+  await deleteAttachments(attachmentList);
+
   return true;
 }
 
+function deleteAttachments(attachmentList) {
+  return Promise.all(attachmentList.map(({ id }) => S3.deleteFile(process.env.BUCKET_NAME, id)));
+}
```

Comment:
Perhaps:
```js
function deleteAttachment(attachment) {
  return S3.deleteFile(process.env.BUCKET_NAME, attachment.id)
}
```

and used like:
```js
const deleteAttachmentPromises = attachmentList.map(deleteAttachment);
await Promise.all(deleteAttachmentPromises);
```

Reply:
Was wondering if we shouldn't use `Promise.all` in this case. If any of the promises rejects then every other promise will reject as well. Perhaps we should use `allSettled` instead and log if any file on s3 fails to be deleted.